src/utils/hacks.ts

```diff
@@ -28,7 +28,7 @@ export function validatePackageRequirement(input: PlainObject): PackageRequireme
   if (constraint === undefined) {
     constraint = '*'
   } else if (isNumber(constraint)) {
-    constraint = `${constraint}`
+    constraint = `~${constraint}`
```

Comment:
before we were interpreting these as `^` ranges, but semver strictly defines “just numbers” to be `~` ranges